### PR TITLE
Feature/pdf reporting

### PR DESCRIPTION
**Pull request by grasbergerm**
Addresses #52

### CODE REVIEW

File: src/main/java/com/coveros/selenified/OutputFile.java [hunk 1]
```diff
@@ -273,6 +276,42 @@ private void replaceInFile(String oldText, String newText) {
...
+        StringBuilder oldContent = new StringBuilder();
+
+        try (FileReader fr = new FileReader(file); BufferedReader reader = new BufferedReader(fr)) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                oldContent.append(line);
+                oldContent.append("\r\n");
+            }
+        } catch (IOException e) {
+            log.error(e);
```
Comment (msaperst) [inline src/main/java/com/coveros/selenified/OutputFile.java]: can we write a UT to trigger this? Maybe with a missing html file?

File: src/main/java/com/coveros/selenified/OutputFile.java [hunk 2]
```diff
@@ -625,6 +675,24 @@ public void finalizeOutputFile(int testStatus) {
...
+     */
+    private void generatePdf() {
+        String pdfFilename = directory + "/" + filename + ".pdf";
+        try (OutputStream os = new FileOutputStream(pdfFilename)) {
+            PdfRendererBuilder builder = new PdfRendererBuilder();
+            builder.withHtmlContent(getHtmlForPDFConversion(), "file://" + pdfFilename.replaceAll(" ", "%20"));
+            builder.toStream(os);
+            builder.run();
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (msaperst) [inline src/main/java/com/coveros/selenified/OutputFile.java]: can we write a UT to trigger this? Maybe with a missing html file?

Comment (msaperst) [reply src/main/java/com/coveros/selenified/OutputFile.java]: this should use the logger

File: src/test/java/unit/OutputFileTest.java [hunk 3]
```diff
@@ -662,4 +662,22 @@ public void formatHTMLSpaceTest() {
...
+    @Test
+    public void generatePDFTest() throws InvalidBrowserException {
+        OutputFile outputFile =
+                new OutputFile("results", "file", new Capabilities(new Browser("Chrome")), null, null, null, null, null, null);
+        File directory = new File("results");
+        File file = new File("results", "file");
+
+        System.setProperty("generatePDF", "true");
+        outputFile.finalizeOutputFile(1);
+        System.clearProperty("generatePDF");
```
Comment (msaperst) [inline src/test/java/unit/OutputFileTest.java]: this isn't a safe way to do it
what if it was set? We're now wiping it out

File: src/main/java/com/coveros/selenified/OutputFile.java [hunk 4]
```diff
@@ -625,6 +674,24 @@ public void finalizeOutputFile(int testStatus) {
...
+        if (System.getProperty("generatePDF") != null && "true".equals(System.getProperty("generatePDF"))) {
+            generatePdf();
+        }
+    }
+
+    /**
+     * Generates a pdf report in the same directory as the html report
+     */
+    private void generatePdf() {
+        String pdfFilename = directory + "/" + filename + ".pdf";
```
Comment (coveros-ci-dev) [inline src/main/java/com/coveros/selenified/OutputFile.java]: MINOR Remove this hard-coded path-delimiter. [rule](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1075)

**Pull request merged**